packages/app-project/src/components/AuthModal/components/RegisterForm/RegisterForm.js

```diff
@@ -57,16 +59,27 @@ function RegisterForm (props) {
           handleBlur,
           handleSubmit,
           isSubmitting,
-          touched,
-          values,
-          /* and other goodies */
-        }) => (
+          values
+        }) => {
+          const userNameFieldHelp = (values.underageWithParent) ?
+            `${counterpart('RegisterForm.usernameHelp')} ${counterpart('RegisterForm.underageNotRealName')}` :
+            counterpart('RegisterForm.usernameHelp')
+          const privacyAgreementLabel = (values.underageWithParent) ?
+            <Text>{counterpart('RegisterForm.underageConsent')}{' '}(<PrivacyPolicyLink />)</Text> :
+            <Text>{counterpart('RegisterForm.privacyAgreement')}{' '}(<PrivacyPolicyLink />)</Text>
+          const emailListSignUpLabel = (values.underageWithParent) ?
+            <Text>{counterpart('RegisterForm.underageEmail')}</Text> :
+            <Text>{counterpart('RegisterForm.emailListSignUp')}</Text>
+
+          return (
             <Box as='form' onSubmit={handleSubmit} margin={{ top: 'small' }}>
+              {generalError && 
+                <Text color={{ light: 'status-error', dark: 'status-error' }}>{generalError}</Text>}
```

Comment:
This should maybe go outside the form tag. Is it showing error messages from the store, rather than error messages for a particular form input?

Reply:
[MDN suggests adding `role="alert"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/alerts) too, which might announce the error in screenreaders without them having to realise they must go back and read the form again.

Reply:
> Is it showing error messages from the store, rather than error messages for a particular form input?

This is the general error catch for the registration request after validations pass, so it's not from the store. I store any errors from that request in the container component state and display them at top. I'm imagining this might be like 400 or 500 errors, so not a specific input either unless someone somehow is able to attempt the registration on a really old browser that doesn't support the built in validations? Unfortunately if it is a specific input error coming back from the API, the panoptes-javascript-client joins the error array into a string which makes it really hard to parse. Notice the regex parsing I'm doing because of it for the other errors :(

Would you prefer we store these kinds of errors in the user store?

Reply:
No, I was just wondering what the `generalError` prop referred to. Sounds like it should definitely be marked up as an alert, outside the form tag.

Reply:
Thinking about it, isn’t there a Grommet component for alerts/notifications in screenreaders? I can’t remember its name but I’m sure I’ve seen it.

Reply:
Here it is: https://v2.grommet.io/announcecontext

Reply:
I'll try that out on another pass at this, but for now I've added the `role="alert"` and moved it outside of the form.

Reply:
No problem. I'm not even sure if that Grommet component does what I thought it did (marks a text element as a live region.)